src/AutoRest.CSharp/MgmtTest/Generation/ResourceCollectionTestWriter.cs

```diff
@@ -236,22 +240,34 @@ protected void WriteMethodTest(MgmtClientOperation clientOperation, bool async,
                     }
 
                     WriteTestDecorator();
-                    var testCaseSuffix = exampleIdx > 0 ? (exampleIdx + 1).ToString() : String.Empty;
+                    var testCaseSuffix = exampleIdx > 0 ? (exampleIdx + 1).ToString() : string.Empty;
                     _writer.Append($"public {GetAsyncKeyword(async)} {MgmtBaseTestWriter.GetTaskOrVoid(async)} {methodName}{testCaseSuffix}()");
                     using (_writer.Scope())
                     {
                         _writer.Line($"// Example: {exampleModel.Name}");
-                        List<KeyValuePair<string, FormattableString>> parameterValues = WriteOperationParameters(clientOperation.MethodParameters, exampleModel);
-                        _writer.Line();
-                        WriteGetCollection(parentTp, operation.RequestPath.SerializedPath, exampleModel, parameterValues);
-                        WriteMethodTestInvocation(async, clientOperation, isLroOperation, $"collection.{testMethodName}", parameterValues.Select(pv => pv.Value));
+                        WriteOperationInvocation(This, clientOperation, operation, exampleModel, async, isLroOperation);
                     }
                     _writer.Line();
                     exampleIdx++;
                 }
             }
         }
 
+        public bool WriteOperationInvocation(ResourceCollection collection, MgmtClientOperation clientOperation, MgmtRestOperation restOperation, ExampleModel exampleModel, bool async, bool isLroOperation)
```

Comment:
I see the same function name in extension test writer, please consider make this abstract or virtual (depending on how much these two have in common) and put it in the base writer and override it if necessary here

Reply:
adding to MgmtBaseTestWriter as abstract